src/common/selectors/index.js

```diff
@@ -1,12 +1,65 @@
 import { createSelector } from 'reselect';
 
+import { parseGitHubRepoUrl } from '../helpers/github-url';
+
 const getSnaps = state => state.snaps;
+const getSnapBuilds = state => state.snapBuilds;
+
+// returns bool - true if there are any snaps in state
+export const hasSnaps = createSelector(
+  [getSnaps],
+  (snaps) => snaps.snaps ? snaps.snaps.length > 0 : false
+);
 
+// returns bool - true if there are no snaps in state
+export const hasNoSnaps = createSelector(
+  [getSnaps],
+  (snaps) => snaps.snaps ? snaps.snaps.length === 0 : true
+);
+
+// returns bool - true if there are no snaps with registered name in state
 export const hasNoRegisteredNames = createSelector(
   [getSnaps],
   (snaps) => {
-    return !snaps.snaps.some((snap) => {
+    return snaps.snaps ? !snaps.snaps.some((snap) => {
       return snap.store_name;
+    }) : true;
+  }
+);
+
+// returns array - snaps that have both registered name and snapcraft data
+export const snapsWithRegisteredNameAndSnapcraftData = createSelector(
```

Comment:
Could we decompose to `getSnapsWithRegisteredName` and `getSnapsWithSnapcraftData` and then use those 2 selectors to compose this and more ... or just leave it as those 2?